src/conn/conn_open.c

```diff
@@ -78,7 +78,7 @@ __wt_connection_close(WT_CONNECTION_IMPL *conn)
      * wiredtiger_open hits an error (as well as during normal shutdown). Assert they're not
      * running.
      */
-    WT_ASSERT(session, !F_ISSET(conn, WT_CONN_SERVER_ASYNC | WT_CONN_SERVER_LSM));
+    WT_ASSERT(session, !FLD_ISSET(conn->server_flags, WT_CONN_SERVER_ASYNC | WT_CONN_SERVER_LSM));
```

Comment:
this '|' is not present in the develop branch, see https://github.com/wiredtiger/wiredtiger/pull/6264/files
Should we leave it or align it with the develop branch ?

Reply:
Leave it as is, the flag `WT_CONN_SERVER_ASYNC` was removed in WT-6674 which hasn't been backported to 4.2.

Reply:
That also means you need to change all uses of `WT_CONN_SERVER_ASYNC` in the `src/async` directory.